[new-product] Eclipse Jetty (#4122)
captn3m0/endoflife.date: products/eclipse-jetty.md

title: Eclipse Jetty
category: server-app
tags: eclipse web-server
iconSlug: eclipsejetty

> permalink: /eclipse-jetty

Reviewer: ```suggestion
permalink: /eclipse-jetty
alternate_urls:
-   /jetty
```

changelogTemplate: https://github.com/jetty/jetty.project/releases/tag/jetty-__LATEST__
releaseDateColumn: true
eolColumn: "Community Support"
extendedSupportColumn: "Extended Support"

identifiers:
-   purl: pkg:maven/org.eclipse.jetty/jetty-server

auto:
-   maven: org.eclipse.jetty/jetty-server  

# eol status and minJavaVersion can be found on https://eclipse.dev/jetty/download.php
releases:
-   releaseCycle: "12"
    releaseDate: 2023-08-07
    eol: false
    extendedSupport: true
    minJavaVersion: 17
    latest: "12.0.3"
    latestReleaseDate: 2023-10-30

# https://github.com/jetty/jetty.project/issues/10485
-   releaseCycle: "11"
    releaseDate: 2020-12-07
    eol: 2024-01-01
    extendedSupport: true
    minJavaVersion: 11
    latest: "11.0.18"
    latestReleaseDate: 2023-10-30

# https://github.com/jetty/jetty.project/issues/10485
-   releaseCycle: "10"
    releaseDate: 2020-12-07
    eol: 2024-01-01
    extendedSupport: true
    minJavaVersion: 11
    latest: "10.0.18"
    latestReleaseDate: 2023-10-30

# https://github.com/jetty/jetty.project/issues/7958
-   releaseCycle: "9.4"
    releaseDate: 2016-12-07
    eol: 2022-06-01

>     extendedSupport: true

Reviewer: Are all versions released after the end of community support made available freely to everybody ? If yes I think this must be documented in the description.

Author: That's not clear to me @marcwrobel . From what I understand on these pages (https://github.com/jetty/jetty.project/issues/7958) the security releases are still available to the public. I'm unsure if the releases which are triggered by webtide customers are publically available.

Reviewer: I figured it out, support is divided into two phases:

- community support : bug and security fixes,
- security support: critical bug and security fixes.

When a release is in the security support phase, extended support is still available if needed, and looks like some changes are backported to publicly available version, such as https://github.com/jetty/jetty.project/releases/tag/jetty-9.3.30.v20211001.

I updated the page accordingly, and did not detailed too much the commercial support as I still don't know if all sponsored changes are made available to everybody or not.

    minJavaVersion: 8
    latest: "9.4.53.v20231009"
    latestReleaseDate: 2023-10-10

-   releaseCycle: "9.3"
    releaseDate: 2015-06-01
    eol: 2022-12-31

>     extendedSupport: 2021-10-01

Reviewer: It's strange to have extended support ending before eol. Is it OK ?

Author: Correct, but maybe we better put it on `false`, since it is unclear / unknown.

We agreed to put the end-of-life date based on the public downloads page, which doesn't stipulate a specific date, but we took 31/12 of that year to have an EOL date. For the recent release (9.4 and later) there is a GitHub issue which documents the "end of community support", which I considered as the "end-of-life" date here, and the effective date on the downloads page as the "end of extended support".

I've changed it in the merge request to `false` for those we don't know.

    eol: 2014-12-31
    extendedSupport: 2016-01-12
    minJavaVersion: 7
    latest: "9.1.6.v20160112"
    latestReleaseDate: 2016-01-12

-   releaseCycle: "9.0"
    releaseDate: 2013-03-08
    eol: 2013-12-31
    extendedSupport: 2013-11-07
    minJavaVersion: 7
    latest: "9.0.7.v20131107"
    latestReleaseDate: 2013-11-07

-   releaseCycle: "8"
    releaseDate: 2011-09-01
    eol: 2014-12-31
    extendedSupport: 2016-09-08
    minJavaVersion: 6
    latest: "8.2.0.v20160908"
    latestReleaseDate: 2016-09-08

-   releaseCycle: "7"
    releaseDate: 2009-10-05
    eol: 2014-12-31
    extendedSupport: 2016-09-08
    minJavaVersion: 5
    latest: "7.6.21.v20160908"
    latestReleaseDate: 2016-09-08

---

> Eclipse Jetty is an open-source (Apache-2.0 licensed) pure-Java HTTP web server environment. It
> implements various [Jakarta web specifications](https://projects.eclipse.org/projects/ee4j.jakartaee-platform),
> including [JSP](https://projects.eclipse.org/projects/ee4j.jsp "Jakarta Server Pages"),
> [Servlets](https://projects.eclipse.org/projects/ee4j.servlet "Jakarta Servlet"),
> and [WebSockets](https://projects.eclipse.org/projects/ee4j.websocket "Jakarta WebSocket").

> Eclipse Jetty does not have a clearly defined support policy. Major releases appear to be supported for approx 5 years.

Reviewer: ```suggestion
Eclipse Jetty does not have a fixed release and support policy. But a list of all versions, along with their support status, is available [on their website](https://eclipse.dev/jetty/download.php).
```

I don't agree with _Major releases appear to be supported for approx 5 years_. I think that major releases support duration varies too much: 9.x is supported since 2013 (10+ years), 7 was supported for six years, 6 and 5 only four years. WDYT of this formulation ?

Author: Looks better to me :).

> | Action                                       | Community Support | Extended Support |
> |----------------------------------------------|-------------------| ---------------- |
> | Community PRs reviewed and integrated        | &check;           | &cross;          |
> | Webtide Customer PRs reviewed and integrated | &check;           | &check;          |
> | Community triggered releases                 | &check;           | &cross;          |
> | Webtide Customer triggered releases          | &check;           | &check;          |
> | Security / Vulnerability triggered releases  | &check;           | &check;          |

Reviewer: ```suggestion
Commercial extended support by the developers of Jetty is available through [Webtide Lifecycle Support](https://webtide.com/).
```

We usually document extended support by only adding a small paragraph stating that it is available and linking to more information (we try to avoid making too much advertising).
